pkg/auth/providers/ldap/ldap_provider.go

```diff
@@ -141,6 +145,46 @@ func (p *ldapProvider) GetPrincipal(principalID string, token v3.Token) (v3.Prin
 		return v3.Principal{}, err
 	}
 
+	if p.samlSearchProvider() {
+		lConn, err := ldap.Connect(config, caPool)
+		if err != nil {
+			return v3.Principal{}, nil
+		}
+		defer lConn.Close()
+
+		if scope == p.userScope {
+			userPrincipals, err := p.searchUser(externalID, config, lConn)
```

Comment:
I dont think leveraging searchUser is really the right thing to be doing here. That function is for fuzzy searching, not for hard lookups, which this is. You should use something similar to the look up we do as part of login:

https://github.com/rancher/rancher/blob/7330335a207c6c0ab98d0b8606a5fdf8cde5bd12/pkg/auth/providers/ldap/ldap_client.go#L46-L49